dbt/adapters/synapse/synapse_relation.py

```diff
@@ -0,0 +1,19 @@
+from dataclasses import dataclass, field
+from typing import Optional, Type
+
+from dbt.adapters.base.relation import BaseRelation, Policy
+from dbt.adapters.utils import classproperty
+
+from dbt.adapters.synapse.relation_configs import SynapseIncludePolicy, SynapseQuotePolicy, SynapseRelationType
+
+
+@dataclass(frozen=True, eq=False, repr=False)
+class SynapseRelation(BaseRelation):
+    type: Optional[SynapseRelationType] = None  # type: ignore
+    quote_policy: SynapseQuotePolicy = field(default_factory=lambda: SynapseQuotePolicy())
+    include_policy: Policy = field(default_factory=lambda: SynapseIncludePolicy())
```

Comment:
I only see two differences between my version and your version.
- I type annotated mine as `SynapseIncludePolicy` instead of just `Policy`
- I inherited from `FabricRelation` instead of `BaseRelation`
I updated my branch to do both of these, and `render` still works as expected. So there might be more to it than just the include policy. I know we mess with the include policies in very specific scenarios. Perhaps that's happening here somewhere.

Reply:
Nevermind, I found it, see above.